src/python/parser.py

```diff
@@ -97,6 +99,23 @@ def parse_function(self):
             ftype = self.parse_type()
         contents = self.parse_block()
         return Node(FUNCTION, [ftype, fname, ptypes, pnames, contents])
+    
+    def parse_struct(self):
+        self.consume(T.STRUCT)
+        sname = self.consume(T.IDENTIFIER)
+        contents = self.parse_struct_block()
+        return Node(STRUCT, contents)
+    
+    def parse_struct_block(self):
+        fields = []
+        self.consume(T.LEFT_CURLY)
+        while not self.consume(T.RIGHT_CURLY) and self.cur < len(self.tokens):
+            fields.append(self.parse_field_declr())
```

Comment:
There is no check for the separators here?
Also, the length check isn't needed as that is handled by `self.consume()`.